libsql-server/src/replication/primary/logger.rs

```diff
@@ -552,7 +552,16 @@ impl ReplicationLogger {
 
         let (new_frame_notifier, _) = watch::channel(generation_start_frame_no);
         unsafe {
-            let conn = rusqlite::Connection::open(db_path.join("data"))?;
```

Comment:
I realize this is a bit scary, why doesn't this conn taking a wal? Sounds like a bug!

Reply:
hm, probably a preexisting one indeed. I'll add it

Reply:
ah, it is preexisting, there's even a comment stating it's a bug. I think it's also not super trivial and we need to figure out the init order here -- this code checkpoints the db before bottomless init, and we need bottomless initialized for the wal manager to work if I remember correctly. I'll go ahead and merge the vfs fix for now to unblock the Unikraft folks, and create a separate issue for the preexisting bug